### ISSUE TITLE
[C] Tests fail on Windows

### ISSUE THREAD

**Issue by realvictorprm** [mono/Embeddinator-4000#343]
The tests for C pass only partly on windows.

Log is here:
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common.Tests.exe is a Catch v1.5.6 host application.
Run with -? for options

-------------------------------------------------------------------------------
Types.C
-------------------------------------------------------------------------------
d:\development\embeddinator\local - kopie\tests\common\tests.c.cpp(9)
...............................................................................

d:\development\embeddinator\local - kopie\tests\common\tests.c.cpp(26): FAILED:
  REQUIRE( Type_Int64_get_Min() == (-2147483647L - 1) )
with expansion:
  -9223372036854775808 == -2147483648

-------------------------------------------------------------------------------
Constructors.C
-------------------------------------------------------------------------------
d:\development\embeddinator\local - kopie\tests\common\tests.c.cpp(141)
...............................................................................

d:\development\embeddinator\local - kopie\tests\common\tests.c.cpp(158): FAILED:
  REQUIRE( Constructors_AllTypeCode_get_TestResult(all2) == true )
with expansion:
  false == true

-------------------------------------------------------------------------------
FieldsInReference.C
-------------------------------------------------------------------------------
d:\development\embeddinator\local - kopie\tests\common\tests.c.cpp(260)
...............................................................................

d:\development\embeddinator\local - kopie\tests\common\tests.c.cpp(261): FAILED:
  REQUIRE( Fields_Class_get_MaxLong() == 2147483647L )
with expansion:
  9223372036854775807 == 2147483647

===============================================================================
test cases:  12 |  9 passed | 3 failed
assertions: 102 | 99 passed | 3 failed
```

Tested through using my PR #278

**Comment by tritao** [comment]
`LONG_MIN` is 32-bits on Windows, which is why it fails.

**Comment by tritao** [comment]
See http://en.cppreference.com/w/cpp/types/integer, `INT64_MIN` and friends.

**Comment by realvictorprm** [comment]
fixed in #278
